src/Helpers/ResponseParser.php

```diff
@@ -0,0 +1,88 @@
+<?php
+declare(strict_types=1);
+
+namespace Eonx\TestUtils\Helpers;
+
+use Eonx\TestUtils\DataTransferObjects\ResponseException;
+use Eonx\TestUtils\Helpers\Exceptions\NoValidResponseException;
+use Eonx\TestUtils\Helpers\Interfaces\ResponseParserInterface;
+use PHPUnit\Framework\Constraint\IsJson;
+use Symfony\Component\HttpFoundation\Response;
+
+class ResponseParser implements ResponseParserInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function parseError(Response $response): ?ResponseException
+    {
+        $content = (string)$response->getContent();
+
+        // if empty string, return early
+        if ($content === '') {
+            return null;
+        }
+
+        $constraint = new IsJson();
+        $isJson = $constraint->evaluate($content, '', true);
```

Comment:
Instead of doing this, lets check the response's content type. Theres no point doing this dance when we already get told what the format is